### DISCUSSION TITLE
VU Meter getValue() Return

### DISCUSSION THREAD

**Discussion by DylanBrianBenton** [tttapa/Control-Surface#368] [Q&A]
Good EVening, 

I see that getValue() for the VU classes returns an integer for 13 possible levels (0-12). Is there documentation on which integer equates to in DBFS? (eg 0 being clipping etc)

**Comment by tttapa** [comment]
AFAIK, there is no standard conversion. The value depends on whatever your DAW sends. The DAW is the one computing the number between 0 and 12, it sends it to the Arduino over MIDI, and Control Surface just gives you back that exact number.

There seems to be quite some differences between different DAWs, especially with the decay rate.

You could try mapping the values yourself. Just load a 1kHz test tone an turn up the gain until the value changes, noting the dB reading of the meters on screen. There are only 12 values, so it shouldn't be too bad.

Some DAWs set the clipping indicator as well, you can retrieve it using the [`VU::getOverload()`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d8/d4c/classMCU_1_1VU.html#ac7697fc99dabb393feae0d4e6bbbf923) method.

**Reply by DylanBrianBenton** [reply]
Thanks mate. Sorry I didn't see the reply. Is there an input for receiving back the volume of a selected track? As in the gain level of the track

**Reply by tttapa** [reply]
You can get the volume of the track using the `PBValue` class on the [`new-input`](https://github.com/tttapa/Control-Surface/tree/new-input/) branch, see [this example](https://tttapa.github.io/Control-Surface-doc/new-input/Doxygen/d0/d6c/Pitch-Bend-Value_8ino-example.html). You can use `PBValue pb {MCU::VOLUME_n};` to get the volume of the `n`-th channel, or `MCU::MASTER_VOLUME` for the master volume. 

AFAIK, the MCU protocol doesn't have a way to report the gain of a track.

**Reply by DylanBrianBenton** [reply]
Thanks mate, just one query that might be stupid, but is there a reason it uses a pitch bend value?

**Reply by tttapa** [reply]
Because that's what the Mackie developers decided to use back in the day, most likely because it's the easiest way to get 14-bit resolution.
